anki/template/template.py

```diff
@@ -200,20 +205,27 @@ def render_unescaped(self, tag_name=None, context=None):
 
     def clozeText(self, txt, ord, type):
         reg = clozeReg
-        if not re.search(reg%ord, txt):
+        if not re.search(reg % ord, txt):
             return ""
+
         def repl(m):
             # replace chosen cloze with type
+            noSound = removeClozeSound(m.group(3))
```

Comment:
Yes, I am aware.

Ok. so the if statement in english reads like this.

if the hint has length when it's sound is removed then write a cloze with the text of the hint and the sound so it will display the hint and play the sound.

Otherwise, if the hint had no length after the sound was removed, display an empty cloze [...] and put the hint which only has a sound in it, along side.

Finally, if the hint had no length at all, just print an empty cloze [...].
   
If it were guaranteed that the hint is not null the elif could be an else, and the last condition could be removed.  Probably a good idea anyway, since removeClozeSound is not going to like getting a null for text.

```
 def repl(m):
            # replace chosen cloze with type
            hint = (m.group(3) if m.group(3) else "")
            answer = m.group(1)
            hintWithNoSound = removeClozeSound(hint)
            if type == "q":
                if hintWithNoSound:
                    return "<span class=cloze>[%s]</span>" % hint
                else:
                    return "<span class=cloze>[...]%s</span>" % hint
            else:
                return "<span class=cloze>%s</span>" % answer
```